flytestdlib/resolver/k8s_resolver.go

```diff
@@ -0,0 +1,177 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/resolver"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/flyteorg/flyte/flytestdlib/logger"
+)
+
+const (
+	K8sSchema = "k8s"
+)
+
+type targetInfo struct {
+	serviceName      string
+	serviceNamespace string
+	port             string
+}
+
+func (t targetInfo) String() string {
+	return fmt.Sprintf("kubernetes:///%s.%s:%s", t.serviceNamespace, t.serviceName, t.port)
+}
+
+// NewBuilder creates a kubeBuilder which is used by grpc resolver.
+func NewBuilder(client kubernetes.Interface, schema string) resolver.Builder {
+	return &kubeBuilder{
+		k8sClient: client,
+		schema:    schema,
+	}
+}
+
+type kubeBuilder struct {
+	k8sClient kubernetes.Interface
+	schema    string
+}
+
+func splitServicePortNamespace(hpn string) (service, port, namespace string) {
+	service = hpn
+
+	colon := strings.LastIndexByte(service, ':')
+	if colon != -1 {
+		service, port = service[:colon], service[colon+1:]
+	}
+
+	parts := strings.SplitN(service, ".", 3)
+	if len(parts) >= 2 {
```

Comment:
Should we set namespace to `default` if parts==1 ?

Reply:
it will use the namespace when agent is at if the namespace is None